### DISCUSSION TITLE
Add `--no-headers` option to `rename`

### DISCUSSION THREAD

**Discussion by jqnatividad** [dathere/qsv#81] [Ideas]
Done in #97.

See below for more details...

**Comment by tmtmtmtm** [comment]
It would be useful if `qsv rename` could take a `--no-headers` argument, as a way to add column names to a CSV that doesn't currently have them. Or is there another way to do this?

**Reply by jqnatividad** [reply]
@tmtmtmtm , you should be able to do this by using `qsv cat columns`, with the first CSV only having the headers.

**Reply by tmtmtmtm** [reply]
@jqnatividad Oh, that's an interesting approach. Is it possible to feed `qsv cat` that column list directly, though, or does this require first creating (and then later deleting) an intermediate file for the headers? If it's only the latter, that seems quite a bit clunkier than, for example, adding them in with `sed`.

**Reply by jqnatividad** [reply]
@tmtmtmtm yes. It does require creating an intermediate file for the headers:

`qsv cat rows headers.csv headerless.csv --no-headers > withheaders.csv`

Actually, you don't even need qsv to do this, and just use `cat`:

`cat headers.csv headerless.csv > withheaders.csv`

So many ways to skin the "cat" :wink:, 

And that's the beauty of a composable CLI - you can mix & match it with other CLI tools like `sed`, `cat`, `curl`, `jq`, etc.  It really depends on your use case and what tools you have access to and familiar with.

For example, for somebody who is creating a data pipeline that receives a headerless csv from a data feed on a regular basis, the `cat` approach is cleaner than using `sed`.

Regardless, if you have a specific use case and a corresponding recipe that uses qsv in some way, please do add it to the [Cookbook](https://github.com/jqnatividad/qsv/wiki/Cookbook#cookbook)!

**Reply by tmtmtmtm** [reply]
@jqnatividad I'm trying to avoid needing to create (and remember to delete again) temporary files. I certainly _can_ use other commands to do this, but one of the reasons I've switched from `xsv` to `qsv` is precisely because I can do now lots of other things directly that I was previously going out to other commands for. `qsv rename` is so much nicer for the cases where the data already has a header row, so I was hoping it would be easy to extend that to also work in the cases where it doesn't, especially as there's already a `--no-headers` option to lots of the other subcommands, that seems like it would also work perfectly here.

**Reply by jqnatividad** [reply]
The same reason I created the `qsv` fork, as I saw so many pending PRs that I needed myself and "scratched the itch."

Anyways, I see your point regarding `--no-headers`, and I do want to make `qsv` as logical, consistent and symmetric as possible, so I'll add it to `rename`.

**Reply by jqnatividad** [reply]
Done. See https://github.com/jqnatividad/qsv/issues/97
